diff --git PKGBUILD PKGBUILD
index e28fbdc..7907549 100644
--- PKGBUILD
+++ PKGBUILD
@@ -8,7 +8,7 @@ url="https://www.gnu.org/software/gettext/"
 arch=(x86_64)
 license=('GPL-2.0-only' 'LGPL-2.0-only' 'GFDL-1.2-only' 'GPL-2.0-or-later')
 depends=(gcc-libs acl sh libunistring libxml2)
-makedepends=(gettext emacs git)
+makedepends=(gettext git)
 optdepends=('git: for autopoint infrastructure updates')
 options=(!docs)
 source=(https://ftp.gnu.org/pub/gnu/gettext/$pkgname-$pkgver.tar.gz{,.sig}
