diff --git PKGBUILD PKGBUILD
index e3867dd..8ec1a46 100644
--- PKGBUILD
+++ PKGBUILD
@@ -25,11 +25,11 @@ pkgver() {
 }
 
 build() {
-  make PREFIX=/usr DISPATCH=1 -C xxHash
+  make PREFIX=/usr DISPATCH=0 -C xxHash
 }
 
 package() {
-  make PREFIX=/usr DISPATCH=1 DESTDIR="${pkgdir}" -C xxHash install
+  make PREFIX=/usr DISPATCH=0 DESTDIR="${pkgdir}" -C xxHash install
   install -Dm 644 xxHash/LICENSE -t "${pkgdir}"/usr/share/licenses/xxhash
 }
 
