diff --git PKGBUILD PKGBUILD
index 695c72c..ae6c40e 100644
--- PKGBUILD
+++ PKGBUILD
@@ -10,11 +10,12 @@ arch=('x86_64')
 license=('GPL2')
 url="http://linux-pam.org"
 depends=('glibc' 'libtirpc' 'pambase' 'audit' 'libaudit.so' 'libxcrypt' 'libcrypt.so')
-makedepends=('flex' 'w3m' 'docbook-xml>=4.4' 'docbook-xsl')
+makedepends=('flex' 'docbook-xml>=4.4' 'docbook-xsl')
 provides=('libpam.so' 'libpamc.so' 'libpam_misc.so')
 backup=(etc/security/{access.conf,faillock.conf,group.conf,limits.conf,namespace.conf,namespace.init,pwhistory.conf,pam_env.conf,time.conf} etc/environment)
 source=(https://github.com/linux-pam/linux-pam/releases/download/v$pkgver/Linux-PAM-$pkgver{,-docs}.tar.xz{,.asc}
-        $pkgname.tmpfiles)
+        $pkgname.tmpfiles
+        pam-change-prefix.patch)
 validpgpkeys=(
         '8C6BFD92EE0F42EDF91A6A736D1A7F052E5924BB' # Thorsten Kukuk
         '296D6F29A020808E8717A8842DB5BD89A340AEB7' #Dimitry V. Levin <ldv@altlinux.org>
@@ -24,15 +25,24 @@ sha256sums=('7ac4b50feee004a9fa88f1dfd2d2fa738a82896763050cd773b3c54b0a818283'
             'SKIP'
             'fe7493aa0a281f8cfe81814768329f953098d0fd8073da1dc0bd64494d022d4d'
             'SKIP'
-            '5631f224e90c4f0459361c2a5b250112e3a91ba849754bb6f67d69d683a2e5ac')
+            '11818458b060ac4d95b2add3a8fc0029ffe91ad71b5b3c0596275f102a2ecc99'
+            'cc0382b570cb813f8ece45b2beba52f7ffa4dd5cb6dd9d290d97a6cd67cb889c')
 
 options=('!emptydirs')
 
+prepare() {
+  cd Linux-PAM-$pkgver
+  patch -Np2 -i ../pam-change-prefix.patch
+}
+
 build() {
   cd Linux-PAM-$pkgver
   ./configure \
+    --prefix=/usr \
     --libdir=/usr/lib \
     --sbindir=/usr/bin \
+    --sysconfdir=/etc \
+    --includedir=/usr/include/security \
     --enable-logind \
     --disable-db
   make
diff --git pam.tmpfiles pam.tmpfiles
index f6e490b..b5c735c 100644
--- pam.tmpfiles
+++ pam.tmpfiles
@@ -1 +1 @@
-d /run/faillock 0755 root root -
+d /data/var/run/faillock 0755 root root -
