diff --git PKGBUILD PKGBUILD
index bed8373..70d3ccc 100644
--- PKGBUILD
+++ PKGBUILD
@@ -11,7 +11,7 @@ arch=(x86_64)
 url="https://pandas.pydata.org/"
 license=(BSD)
 depends=('python-numpy' 'python-dateutil' 'python-pytz')
-makedepends=('cython' 'python-setuptools')
+makedepends=('cython0' 'python-setuptools')
 optdepends=(
     'python-pandas-datareader: pandas.io.data replacement (recommended)'
     'python-numexpr: accelerating certain numerical operations (recommended)'
