diff --git PKGBUILD PKGBUILD
index 7ee1bce..cdb2fe1 100644
--- PKGBUILD
+++ PKGBUILD
@@ -12,7 +12,7 @@ pkgdesc='Miscellaneous system utilities for Linux'
 url='https://github.com/util-linux/util-linux'
 arch=('x86_64')
 makedepends=('git' 'meson' 'asciidoctor' 'bash-completion' 'libcap-ng'
-             'libutempter' 'libxcrypt' 'python' 'systemd')
+             'libutempter' 'libxcrypt' 'python')
 license=(
   'BSD-2-Clause'
   'BSD-3-Clause'
@@ -32,7 +32,8 @@ source=("git+https://github.com/util-linux/util-linux#tag=${_tag}?signed"
         'util-linux.sysusers'
         '60-rfkill.rules'
         'rfkill-unblock_.service'
-        'rfkill-block_.service')
+        'rfkill-block_.service'
+        'change-pathnames.patch')
 sha256sums=('SKIP'
             '6ffedbc0f7878612d2b23589f1ff2ab15633e1df7963a5d9fc750ec5500c7e7a'
             'ee917d55042f78b8bb03f5467e5233e3e2ddc2fe01e302bc53b218003fe22275'
@@ -43,7 +44,8 @@ sha256sums=('SKIP'
             '10b0505351263a099163c0d928132706e501dd0a008dac2835b052167b14abe3'
             '7423aaaa09fee7f47baa83df9ea6fef525ff9aec395c8cbd9fe848ceb2643f37'
             '8ccec10a22523f6b9d55e0d6cbf91905a39881446710aa083e935e8073323376'
-            'a22e0a037e702170c7d88460cc9c9c2ab1d3e5c54a6985cd4a164ea7beff1b36')
+            'a22e0a037e702170c7d88460cc9c9c2ab1d3e5c54a6985cd4a164ea7beff1b36'
+            'aa72f8584f5962d3fae944a82853571f94670d3aa0abbbf57bbe2a58e6e62b93')
 
 _backports=(
 )
@@ -66,18 +68,26 @@ prepare() {
     git revert --mainline 1 --no-commit "${_c}"
   done
 
+  patch -Np1 -i ../change-pathnames.patch
+
   # do not mark dirty
   sed -i '/dirty=/c dirty=' tools/git-version-gen
+  sed -i "s@'/run'@'/data/var/run'@" meson.build
+
+  sed -i '1s|.*|#!/data/usr/bin/sh|' tools/{ko-release-{gen,push},*.sh,config-gen}
 }
 
 build() {
   local _meson_options=(
-    -Dfs-search-path=/usr/bin:/usr/local/bin
+    -Dfs-search-path=/data/usr/bin:/data/usr/local/bin
 
     -Dlibuser=disabled
     -Dncurses=disabled
     -Dncursesw=enabled
     -Deconf=disabled
+    -Dsystemd=disabled
+    -Dcryptsetup=disabled
+    -Dcryptsetup-dlopen=disabled
 
     -Dbuild-chfn-chsh=enabled
     -Dbuild-line=disabled
@@ -96,65 +106,53 @@ package_util-linux() {
   conflicts=('rfkill' 'hardlink')
   provides=('rfkill' 'hardlink')
   replaces=('rfkill' 'hardlink')
-  depends=('glibc' 'pam' 'shadow' 'coreutils' 'systemd-libs' 'libsystemd.so'
-           'libudev.so' 'libcap-ng' 'libutempter' 'libxcrypt' 'libcrypt.so' 'util-linux-libs'
+  depends=('glibc' 'pam' 'shadow' 'coreutils'
+           'libcap-ng' 'libutempter' 'libxcrypt' 'libcrypt.so' 'util-linux-libs'
            'libmagic.so' 'libncursesw.so' 'readline' 'zlib')
   optdepends=('words: default dictionary for look')
-  backup=(etc/pam.d/chfn
-          etc/pam.d/chsh
-          etc/pam.d/login
-          etc/pam.d/remote
-          etc/pam.d/runuser
-          etc/pam.d/runuser-l
-          etc/pam.d/su
-          etc/pam.d/su-l)
+  backup=(data/etc/pam.d/chfn
+          data/etc/pam.d/chsh
+          data/etc/pam.d/login
+          data/etc/pam.d/remote
+          data/etc/pam.d/runuser
+          data/etc/pam.d/runuser-l
+          data/etc/pam.d/su
+          data/etc/pam.d/su-l)
 
   _python_stdlib="$(python -c 'import sysconfig; print(sysconfig.get_paths()["stdlib"])')"
 
   DESTDIR="${pkgdir}" meson install -C build
 
   # remove static libraries
-  rm "${pkgdir}"/usr/lib/lib*.a*
+  rm "${pkgdir}"/data/usr/lib/lib*.a*
 
   # setuid chfn and chsh
-  chmod 4755 "${pkgdir}"/usr/bin/{newgrp,ch{sh,fn}}
+  chmod 4755 "${pkgdir}"/data/usr/bin/{newgrp,ch{sh,fn}}
 
   # install PAM files for login-utils
-  install -Dm0644 pam-common "${pkgdir}/etc/pam.d/chfn"
-  install -m0644 pam-common "${pkgdir}/etc/pam.d/chsh"
-  install -m0644 pam-login "${pkgdir}/etc/pam.d/login"
-  install -m0644 pam-remote "${pkgdir}/etc/pam.d/remote"
-  install -m0644 pam-runuser "${pkgdir}/etc/pam.d/runuser"
-  install -m0644 pam-runuser "${pkgdir}/etc/pam.d/runuser-l"
-  install -m0644 pam-su "${pkgdir}/etc/pam.d/su"
-  install -m0644 pam-su "${pkgdir}/etc/pam.d/su-l"
+  install -Dm0644 pam-common "${pkgdir}/data/etc/pam.d/chfn"
+  install -m0644 pam-common "${pkgdir}/data/etc/pam.d/chsh"
+  install -m0644 pam-login "${pkgdir}/data/etc/pam.d/login"
+  install -m0644 pam-remote "${pkgdir}/data/etc/pam.d/remote"
+  install -m0644 pam-runuser "${pkgdir}/data/etc/pam.d/runuser"
+  install -m0644 pam-runuser "${pkgdir}/data/etc/pam.d/runuser-l"
+  install -m0644 pam-su "${pkgdir}/data/etc/pam.d/su"
+  install -m0644 pam-su "${pkgdir}/data/etc/pam.d/su-l"
 
   # TODO(dreisner): offer this upstream?
-  sed -i '/ListenStream/ aRuntimeDirectory=uuidd' "${pkgdir}/usr/lib/systemd/system/uuidd.socket"
+  # sed -i '/ListenStream/ aRuntimeDirectory=uuidd' "${pkgdir}/data/usr/lib/systemd/system/uuidd.socket"
 
   # runtime libs are shipped as part of util-linux-libs
   install -d -m0755 util-linux-libs/lib/
-  mv "$pkgdir"/usr/lib/lib*.so* util-linux-libs/lib/
-  mv "$pkgdir"/usr/lib/pkgconfig util-linux-libs/lib/pkgconfig
-  mv "$pkgdir"/usr/include util-linux-libs/include
+  mv "$pkgdir"/data/usr/lib/lib*.so* util-linux-libs/lib/
+  mv "$pkgdir"/data/usr/lib/pkgconfig util-linux-libs/lib/pkgconfig
+  mv "$pkgdir"/data/usr/include util-linux-libs/include
   mv "$pkgdir"/"${_python_stdlib}"/site-packages util-linux-libs/site-packages
   rmdir "$pkgdir"/"${_python_stdlib}"
-  mv "$pkgdir"/usr/share/man/man3 util-linux-libs/man3
-
-  # install systemd-sysusers
-  install -Dm0644 util-linux.sysusers \
-    "${pkgdir}/usr/lib/sysusers.d/util-linux.conf"
-
-  install -Dm0644 60-rfkill.rules \
-    "${pkgdir}/usr/lib/udev/rules.d/60-rfkill.rules"
-
-  install -Dm0644 rfkill-unblock_.service \
-    "${pkgdir}/usr/lib/systemd/system/rfkill-unblock@.service"
-  install -Dm0644 rfkill-block_.service \
-    "${pkgdir}/usr/lib/systemd/system/rfkill-block@.service"
+  mv "$pkgdir"/data/usr/share/man/man3 util-linux-libs/man3
 
-  install -vDm 644 $pkgbase/Documentation/licenses/COPYING.{BSD*,ISC} -t "$pkgdir/usr/share/licenses/$pkgname/"
-  install -vDm 644 $pkgbase-BSD-2-Clause.txt -t "$pkgdir/usr/share/licenses/$pkgname/"
+  install -vDm 644 $pkgbase/Documentation/licenses/COPYING.{BSD*,ISC} -t "$pkgdir/data/usr/share/licenses/$pkgname/"
+  install -vDm 644 $pkgbase-BSD-2-Clause.txt -t "$pkgdir/data/usr/share/licenses/$pkgname/"
 }
 
 package_util-linux-libs() {
@@ -165,12 +163,12 @@ package_util-linux-libs() {
   replaces=('libutil-linux')
   optdepends=('python: python bindings to libmount')
 
-  install -d -m0755 "$pkgdir"/{"${_python_stdlib}",usr/share/man/}
-  mv util-linux-libs/lib/* "$pkgdir"/usr/lib/
-  mv util-linux-libs/include "$pkgdir"/usr/include
+  install -d -m0755 "$pkgdir"/{"${_python_stdlib}",data/usr/share/man}
+  mv util-linux-libs/lib/* "$pkgdir"/data/usr/lib/
+  mv util-linux-libs/include "$pkgdir"/data/usr/include
   mv util-linux-libs/site-packages "$pkgdir"/"${_python_stdlib}"/site-packages
-  mv util-linux-libs/man3 "$pkgdir"/usr/share/man/man3
+  mv util-linux-libs/man3 "$pkgdir"/data/usr/share/man/man3
 
-  install -vDm 644 $pkgbase/Documentation/licenses/COPYING.{BSD*,ISC} -t "$pkgdir/usr/share/licenses/$pkgname/"
-  install -vDm 644 $pkgbase-BSD-2-Clause.txt -t "$pkgdir/usr/share/licenses/$pkgname/"
+  install -vDm 644 $pkgbase/Documentation/licenses/COPYING.{BSD*,ISC} -t "$pkgdir/data/usr/share/licenses/$pkgname/"
+  install -vDm 644 $pkgbase-BSD-2-Clause.txt -t "$pkgdir/data/usr/share/licenses/$pkgname/"
 }
